src/Pelagos/Bundle/AppBundle/Controller/DownloadController.php

```diff
@@ -115,4 +115,24 @@ public function httpAction($id)
 
         return $response;
     }
+
+    /**
+     * Get the dataset required information for the Download dialog box.
+     *
+     * @param Dataset $dataset A dataset instance.
+     *
+     * @return array
+     */
+    private function getDatasetDetails(Dataset $dataset): array
+    {
+        $datasetSubmission = $dataset->getDatasetSubmission();
+        return array(
```

Comment:
What happens if the submission is null, as would be the case in a DIF only?

Reply:
> What happens if the submission is null, as would be the case in a DIF only?

Is should probably get caught, but there is nothing to download with a DIF, so this function won't get called. The download button is grayed out.

Reply:
i'll add a check if it is datasetSubmission object, just to be sure